appengine/express-memcached-session/server.js

```diff
@@ -1,64 +1,69 @@
-// Copyright 2015-2016, Google, Inc.
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//    http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
+/**
+ * Copyright 2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
 // [START app]
 'use strict';
 
 // [START setup]
-var express = require('express');
-var session = require('express-session');
-var cookieParser = require('cookie-parser');
-var MemcachedStore = require('connect-memcached')(session);
-var publicIp = require('public-ip');
-var crypto = require('crypto');
+const express = require('express');
+const session = require('express-session');
+const cookieParser = require('cookie-parser');
+const MemcachedStore = require('connect-memcached')(session);
+const publicIp = require('public-ip');
+const crypto = require('crypto');
+let MEMCACHE_URL = process.env.MEMCACHE_URL;
+if (process.env.GAE_MEMCACHE_HOST && process.env.GAE_MEMCACHE_PORT) {
```

Comment:
I didn't want to do that because `GAE_MEMCACHE_*` are environment variables that are specific to App Engine Flexible Environment. If the user wants to connect to an entirely different memcache instance (not Google's) then it seems weird that they'd need to set `GAE_MEMCACHE_*` env vars. Therefore I made the simpler, provider-agnostic `MEMCACHE_URL` env var.